optimize/functions/vlse.go

```diff
@@ -66,6 +112,30 @@ func (CrossInTray) Func(x []float64) float64 {
 	return -0.0001 * math.Pow(math.Abs(math.Sin(x0)*math.Sin(x1)*math.Exp(exp))+1, 0.1)
 }
 
+// DixonPrice implements the DixonPrice function, a function of arbitrary dimension
+// Its typical domain is the hypercube of [-10, 10]^d.
+// The function is given by
+//  f(x) = (x_0-1)^2 + \sum_{i=1}^{d-1} (i+1) * (2*x_i^2-x_{i-1})^2
+// where d is the input dimension. There is a single global minimum, which has
+// a value of
+//  f(x) = 0
+//  x_i = 2^{-(2^{i-1}-2)/(2^{i-1})} for i = 0, ..., d-1.
+// Reference:
+//  https://www.sfu.ca/~ssurjano/dixonpr.html (obtained June 2017)
+type DixonPrice struct{}
+
+func (DixonPrice) Func(x []float64) float64 {
+	xp := x[0]
+	v := (xp - 1) * (xp - 1)
```

Comment:
Can you be certain the slice is not empty?

Reply:
No, but the function is undefined if the input length is 0. I don't think it's unreasonable to panic in that case.

Reply:
Probably could have a better message though.

Reply:
My golang naivete is showing.

Reply:
Not golang so much as Gonum. Our policy is to panic when things are obviously wrong, and this function isn't defined with a 0-D input.